src/api-service/__app__/onefuzzlib/endpoint_authorization.py

```diff
@@ -43,6 +44,38 @@ def is_agent(token_data: UserInfo) -> bool:
     return False
 
 
+def can_modify_config(req: func.HttpRequest, config: InstanceConfig) -> bool:
```

Comment:
It seems like this and `can_modify_pools` should just accept a `UserInfo` instead off an `HttpRequest`.

Reply:
The 4 callers do not have a `UserInfo`, but they do have a `HttpRequest`.  Instead of having all 4 callers call parse_jwt_token, do it once here.

Reply:
I see your point about factoring out the JWT inspection. I do think it is being factored into the wrong place; it should go up the stack, closer to the HTTP/auth layer.

That said, if you want to keep this for practical reasons, then that's fine. My suggestion here is still to change the signature, and require the callers to invoke `parse_jwt_token()`, even though it is currently repetitive. I think it will make it easer to do the right thing in later work. Your call.